### PR TITLE
plugins: Store custom keys in "result"

### PR DESCRIPTION
**Pull request by maringuu**
~~WIP as this needs some changes on the tests that I'd like to do after #922~~ - Actually the changes are orthogonal

### CODE REVIEW

File: src/plugins/analysis/architecture_detection/internal/dt.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def _get_compatible_entry(dts: str) -> str | None:
 
 def construct_result(file_object):
     result = {}
-    for dt_dict in file_object.processed_analysis.get('device_tree', {}).get('device_trees', []):
+    for dt_dict in file_object.processed_analysis['device_tree']["result"].get('device_trees', []):
```
[the remark this answers is not in the document]
Comment (maringuu) [reply src/plugins/analysis/architecture_detection/internal/dt.py:76]: > Also what happened to the "double-quote-string-fixer" pre-commit hook?

Muscle memory to ignore it..

**Comment by maringuu** [comment]
Converted to draft again as this still has many bugs.

File: src/unpacker/unpack.py [hunk 2]
```diff
@@ -48,7 +50,16 @@ def unpack(
...
-        current_fo.processed_analysis['unpacker'] = json.loads(Path(tmp_dir, 'reports', 'meta.json').read_text())
+        current_fo.processed_analysis['unpacker'] = sanitize_processed_analysis(
+            json.loads(
+                Path(
+                    tmp_dir,
+                    'reports',
+                    'meta.json',
+                ).read_text(),
+            ),
+        )
```
Comment (jstucke) [inline src/unpacker/unpack.py:61]: Wow this is a rare case where breaking the lines actually worsens the readability :sweat_smile: 
```suggestion
        report = Path(tmp_dir) / 'reports/meta.json'
        current_fo.processed_analysis['unpacker'] = sanitize_processed_analysis(json.loads(report.read_text()))
```

**Review by jstucke — CHANGES_REQUESTED** [review]

File: src/storage/entry_conversion.py [hunk 3]
```diff
@@ -148,6 +142,11 @@ def _sanitize_list(value: list) -> list:
     return value
 
 
+def _sanitize_returning(data: dict) -> dict:
+    sanitize(data)
+    return data
```
[the remark this answers is not in the document]
Comment (maringuu) [reply src/storage/entry_conversion.py:147]: > I'm not sure what the function name is supposed to mean#

It's a returning version of the sanitize.

But yeah I agree, it doesn't hurt to let sanitize return.

[the remark this answers is not in the document]
Comment (maringuu) [reply src/storage/entry_conversion.py:147]: > I'm not sure what the function name is supposed to mean#

It's a returning version of the sanitize.

But yeah I agree, it doesn't hurt to let sanitize return.

**Review by jstucke — APPROVED** [review]
LGTM!

**Pull request merged**